index.bs

```diff
@@ -2714,40 +2714,41 @@ partial interface MLGraphBuilder {
 <div>
     **Arguments:**
         - *input*: an {{MLOperand}}. The input N-D tensor from which the values are gathered.
-        - *indices*: an {{MLOperand}}. The indices N-D tensor of the input values to gather. The values must be of type {{MLOperandDataType/"uint32"}} or {{MLOperandDataType/"int64"}} in the range [0, N-1] where N is the size of the input dimension indexed by *options.axis*.
+        - *indices*: an {{MLOperand}}. The indices N-D tensor of the input values to gather. The values must be of type {{MLOperandDataType/"uint32"}} or {{MLOperandDataType/"int64"}}, and are clamped to the range -N (inclusive) to N (exclusive) where N is the size of the input dimension indexed by *options.axis*, and a negative index means indexing from the end of the dimension.
```

Comment:
```suggestion
        - *indices*: an {{MLOperand}}. The indices N-D tensor of the input values to gather. The values must be of type {{MLOperandDataType/"uint32"}} or {{MLOperandDataType/"int64"}}, and must be to the range -N (inclusive) to N (exclusive) where N is the size of the input dimension indexed by *options.axis*, and a negative index means indexing from the end of the dimension.
```

Since the primary motivation of this clause is really about security mitigation for the case of garbage in, rather than prescribing algorithm behavior (I don't want people to rely on this assumption that bad indices will be automagically clamped for them when they should never pass them in the first place), then I'd write this sentence as more a statement that the indices *should* be limited to that range, and the note below can recommend a mitigation. I agree that clamping to the range -N (inclusive) to N (exclusive) makes sense, but it's also one possible prescribed mitigation for degenerate input. For example, DML ensures that the computed input offset read is within the bounds of the input tensor (so at least no off-memory reads), but not that each index is constrained to the dimensions.